joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala

```diff
@@ -287,8 +273,8 @@ class RecoverForPythonFile(
       // TODO: inheritsFromTypeFullName does not give full name in pysrc2cpg
       val baseTypeFullNames = cpg.typ.nameExact(baseTypes: _*).fullName.toSeq
       (parentTypes ++ baseTypeFullNames)
-        .map(_.concat(".<body>"))
-        .filterNot(t => t.toLowerCase.matches("(any|object)"))
+        .map(f => f + "." + f.split("\\.").last + "<body>")
```

Comment:
This looks like it can throw a `NoSuchElementException` if `f` does not contain a `.`